### PR TITLE
DBAAS-906 [RHODA][R4] Operator Metrics - Instance with Pending status on Thanos loaded on Grafana with status True

### PR DESCRIPTION
**Pull request by jianrongzhang89**
This PR fixes the issue that there is a discrepancy between the DBaaSInstance provisioning phase with its status as shown on grafana when a MongDO DB instance is deleted from MongoDB Atlas portal. 

Root cause: When an instance gets deleted from ISV, the next reconciliation of DBaaSInstance CR will find that the instance is no longer in Atlas, so it sets the CR status with phase "Pending" indicating the instance will get recreated. However the corresponding dependent AtlasDeployment CR's status may remain as the existing value until its reconciliation kicks in and hence cause the discrepancy between the statuses of these CRs.

Solution:
1) If the current MongoDBAtlasInstance is in "Ready" status, but the current cluster provisioning state in Atlas is different (ie, creating etc), that means the current MongoDBAtlasInstance status is out of date and the corresponding AtlasDeployment CR must be reconciled so that it has the correct status.
2) In order to trigger the reconciliation of the AtlasDeployment CR, the MongoDBAtlasInstance controller updates an annotation "atlas.mongodb.com/updated-at" on the AtlasDeploymentCR. For this to work, the AtlasDeployment controller must be initiated with the predicate that allows the annotation update event.

Fix is tested in a local devt env successfully.

### CODE REVIEW

**Comment by jianrongzhang89** [comment]
PR for Mongo Atlas Operator for review: https://github.com/RHEcosystemAppEng/mongodb-atlas-kubernetes/pull/62

**Review by tchughesiv — CHANGES_REQUESTED** [review]
seeing this failure when running unit test -
```shell
% make unit-test
go test -race -cover ./pkg/...
?   	github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/dbaas	[no test files]
ok  	github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1	(cached)	coverage: 3.1% of statements
?   	github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/authmode	[no test files]
?   	github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/common	[no test files]
?   	github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/project	[no test files]
?   	github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/provider	[no test files]
?   	github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status	[no test files]
ok  	github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/atlas	(cached)	coverage: 56.4% of statements
ok  	github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/atlasconnection	0.836s	coverage: 66.5% of statements
ok  	github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/atlasdatabaseuser	(cached)	coverage: 11.5% of statements
ok  	github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/atlasdeployment	(cached)	coverage: 8.2% of statements
--- FAIL: TestSetInstanceStatusWithDeploymentInfo (0.01s)
    --- FAIL: TestSetInstanceStatusWithDeploymentInfo/DeploymentCreating (0.00s)
        atlasinstance_test.go:340: 
            	Error Trace:	atlasinstance_test.go:340
            	Error:      	Not equal: 
            	            	expected: "False"
            	            	actual  : "True"
            	            	
            	            	Diff:
            	            	--- Expected
            	            	+++ Actual
            	            	@@ -1 +1 @@
            	            	-False
            	            	+True
            	Test:       	TestSetInstanceStatusWithDeploymentInfo/DeploymentCreating
FAIL
```

**Comment by jianrongzhang89** [comment]
@tchughesiv Fixed the unit test case.

File: config/manager/manager.yaml [hunk 1]
```diff
@@ -60,5 +60,5 @@ spec:
                 fieldRef:
                   fieldPath: metadata.namespace
             - name: SYNC_PERIOD_MIN
-              value: 180
+              value: "180"
```
Comment (redhatHameed) [inline config/manager/manager.yaml:63]: do required this change ?

File: pkg/controller/atlasinstance/atlasinstance_controller.go [hunk 2]
```diff
@@ -479,7 +502,12 @@ func setInstanceStatusWithDeploymentInfo(atlasClient *mongodbatlas.Client, inst
 		if cond.Type == status.DeploymentReadyType {
 			statusFound = true
 			if cond.Status == corev1.ConditionTrue {
-				dbaas.SetInstanceCondition(inst, dbaasv1alpha1.DBaaSInstanceProviderSyncType, metav1.ConditionStatus(cond.Status), "Ready", cond.Message)
+				if inst.Status.Phase == dbaasv1alpha1.InstancePhaseReady {
+					dbaas.SetInstanceCondition(inst, dbaasv1alpha1.DBaaSInstanceProviderSyncType, metav1.ConditionStatus(cond.Status), "Ready", cond.Message)
+				} else {
```
Comment (redhatHameed) [inline pkg/controller/atlasinstance/atlasinstance_controller.go]: can we add return here,  and remove else WDYT ?

**Review by redhatHameed — COMMENTED** [review]
@jianrongzhang89 added few minor comments

File: config/manager/manager.yaml (hunk 1 above)
Comment (jianrongzhang89) [reply config/manager/manager.yaml:63]: Yes. It fixes the issue found by the linter.

File: pkg/controller/atlasinstance/atlasinstance_controller.go (hunk 2 above)
Comment (jianrongzhang89) [reply pkg/controller/atlasinstance/atlasinstance_controller.go]: Good idea. Done.

**Review by redhatHameed — APPROVED** [review]

**Pull request merged**